topdown/eval.go

```diff
@@ -2924,16 +2924,16 @@ func (q vcKeyScope) Compare(other ast.Value) int {
 	return 1
 }
 
-func (q vcKeyScope) Find(_ ast.Ref) (ast.Value, error) {
+func (vcKeyScope) Find(ast.Ref) (ast.Value, error) {
 	return nil, nil
 }
 
 func (q vcKeyScope) Hash() int {
 	var hash int
-	for _, v := range q.Ref {
+	for i, v := range q.Ref {
 		if _, ok := v.Value.(ast.Var); ok {
 			// all vars are equal
-			hash++
+			hash += i + 1
```

Comment:
Your commit message makes the think it was intentional 😎

Reply:
Yeah. it's different. I had the notion that we should make the hash different if the order of the entries are different. But since we don't take that into consideration for non-var entries, and since the hash map will call `Compare()` on conflicts anyways, I'm backing out of this.